bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/flamingos2/Flamingos2Parameters.java

```diff
@@ -1,63 +1,51 @@
 package edu.gemini.itc.flamingos2;
 
 import edu.gemini.itc.shared.InstrumentDetails;
+import edu.gemini.spModel.gemini.flamingos2.Flamingos2.*;
 
 /**
- * This class holds the information from the Acquisition Camera section
- * of an ITC web page.  This object is constructed from a servlet request.
+ * Flamingos2 parameters relevant for ITC.
  */
 public final class Flamingos2Parameters implements InstrumentDetails {
 
-    // ITC web form input values.
-    // These constants must be kept in sync with the web page form.
-    // They are used to parse form data.
-    public static final String CLEAR = "Open";
-
-    // Grism names
-    public static final String JHGRISM = "JH";
-    public static final String HKGRISM = "HK";
-    public static final String R3KGRISM = "R3K";
-    public static final String NOGRISM = "None";
-
     // Data members
-    private final String _colorFilter;  // U, V, B, ...
-    private final String _grism;
-    private final String _readNoise;
-    private final String _fpMask;
+    private final Filter filter;
+    private final Disperser grism;
+    private final ReadMode readMode;
+    private final FPUnit fpMask;
 
     /**
      * Constructs a AcquisitionCamParameters from a servlet request
      */
-    public Flamingos2Parameters(final String colorFilter, final String grismName, final String fpMask, final String readNoise) {
-        _colorFilter = colorFilter;
-        _grism = grismName;
-        _fpMask = fpMask;
-        _readNoise = readNoise;
+    public Flamingos2Parameters(final Filter filter, final Disperser grism, final FPUnit fpMask, final ReadMode readNoise) {
+        this.filter = filter;
+        this.grism = grism;
+        this.fpMask = fpMask;
+        this.readMode = readNoise;
     }
 
-    public String getColorFilter() {
-        return _colorFilter;
+    public Filter getFilter() {
```

Comment:
This is the main focus of the current work. This will make everything a ton easier.